internal/providers/azure/azure.go

```diff
@@ -112,7 +112,16 @@ func fetchFromIMDS(f *resource.Fetcher) ([]byte, error) {
 	headers := make(http.Header)
 	headers.Set("Metadata", "true")
 
-	data, err := f.FetchToBuffer(imdsUserdataURL, resource.FetchOptions{Headers: headers})
+	// Azure IMDS expects some codes <500 to still be retried...
+	// Here, we match the cloud-init set.
+	// https://github.com/canonical/cloud-init/commit/c1a2047cf291
+	// https://github.com/coreos/ignition/issues/1806
+	retryCodes := []int{
+		404, // not found
+		410, // gone
+		429, // rate-limited
```

Comment:
idea: we should consider respecting the `retry-after` header here so we don't cascade 429 errors.

Reply:
Ideally, yes. We currently do an exponential backoff which will help a lot here at least.